core/storage/question/gae_models.py

```diff
@@ -244,6 +246,32 @@ def get_question_skill_links_and_skill_descriptions(
         )
         return question_skill_link_models, skill_descriptions, next_cursor_str
 
+    @classmethod
+    def get_question_skill_links_with_constant_number_per_skill(
+            cls, question_count, skill_ids):
+        """Fetches the list of constant number of QuestionSkillLinkModels
+        linked to the skills.
+
+        Args:
+            question_count: int. The number of questions to be returned.
+            skill_ids: list(str). The ids of skills for which the linked
+                question ids are to be retrieved.
+
+        Returns:
+            list(QuestionSkillLinkModel). Constant number of
+                QuestionSkillLinkModels corresponding to given skill_ids.
+        """
+        question_count_per_skill = int(
+            math.ceil(float(question_count) / float(len(skill_ids))))
+        question_skill_link_models = []
+        for skill_id in skill_ids:
+            question_skill_link_models.extend(
+                cls.query(cls.skill_id == skill_id).fetch(
+                    question_count_per_skill
+                )
+            )
+        return question_skill_link_models
```

Comment:
Are you actually returning exactly question_count items here, in the case where question_count is not evenly divisible by len(skill_ids)?

Reply:
Done.